pkg/machine/qemu/machine.go

```diff
@@ -384,16 +384,43 @@ func (v *MachineVM) Init(opts machine.InitOptions) (bool, error) {
 		return false, os.WriteFile(v.getIgnitionFile(), inputIgnition, 0644)
 	}
 	// Write the ignition file
-	ign := machine.DynamicIgnition{
+	ign := &machine.DynamicIgnition{
 		Name:      opts.Username,
 		Key:       key,
 		VMName:    v.Name,
+		VMType:    machine.QemuVirt,
 		TimeZone:  opts.TimeZone,
 		WritePath: v.getIgnitionFile(),
 		UID:       v.UID,
 	}
 
-	err = machine.NewIgnitionFile(ign, machine.QemuVirt)
+	if err := ign.GenerateIgnitionConfig(); err != nil {
+		return false, err
+	}
+
+	// ready is a unit file that sets up the virtual serial device
+	// where when the VM is done configuring, it will send an ack
+	// so a listening host knows it can being interacting with it
+	ready := `[Unit]
```

Comment:
Consider moving the two units into a dedicated `.go` file as consts.  Future changes/bug fixes may need to be applied to both units.  If they're located in the same file, the risk of changing one but not the other is probably lower.

Reply:
I was more thinking about locality.  Currently the units are hidden in the functions.  Having them side-by-side may help future development/maintenance but that's more speculation.